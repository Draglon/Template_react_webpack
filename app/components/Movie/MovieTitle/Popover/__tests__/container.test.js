import React from 'react';
import configureStore from 'redux-mock-store';

import PopoverContainer from '../container';

describe('<PopoverContainer />', () => {
  const state = {
    popoverVisible: false,
  };

  const props = {
    movieId: 999,
  };

  const store = configureStore()(state);
  store.dispatch = jest.fn();

  const container = shallow(<PopoverContainer store={store} {...props} />);
  const instance = container.instance();

  describe('handleVisiblePopover()', () => {
    it('should call handleVisiblePopover()', () => {
      instance.handleVisiblePopover(false);
      expect(container.state('popoverVisible')).toEqual(false);
    });
  });

  describe('closePopover()', () => {
    it('should call closePopover()', () => {
      const spy = jest.spyOn(instance, 'handleVisiblePopover');
      instance.closePopover();
      expect(spy).toHaveBeenCalledWith(false);
    });
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
