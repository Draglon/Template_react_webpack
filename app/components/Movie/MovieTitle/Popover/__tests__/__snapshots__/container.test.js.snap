// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopoverContainer /> should match snapshot 1`] = `
<PopoverComponent
  closePopover={[Function]}
  handleVisiblePopover={
    [MockFunction] {
      "calls": Array [
        Array [
          false,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  movieId={999}
  popoverVisible={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
